plugins/callback/influxdb.py

```diff
@@ -0,0 +1,160 @@
+        env:
+          - name: INFLUXDB_VALIDATE_CERT
+        ini:
+          - section: callback_influxdb
+            key: influxdb_validate_cert
+        default: True
+        type: bool
+'''
+
+EXAMPLES = '''
+examples: >
+  To enable, add this to your ansible.cfg file in the defaults block
+    [defaults]
+    callback_whitelist = community.general.influxdb
+
+  Set the environment variable
+    export INFLUXDB_ADDR=localhost:8086 INFLUXDB_DB=test INFLUXDB_TLS=false
+
+  Set the ansible.cfg variable in the callback_influxdb block
+    [callback_influxdb]
+    INFLUXDB_ADDR = localhost:8086
+    INFLUXDB_DB = test
+    INFLUXDB_TLS = false
+'''
+
+import os
+
+from ansible import context
+from datetime import datetime
+
+from ansible.module_utils._text import to_text, to_native
+from ansible.plugins.callback import CallbackBase
+from ansible.module_utils.urls import open_url
+
+
+class CallbackModule(CallbackBase):
+    """
+    This callback module adds play duration to InfluxDB
+    """
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'aggregate'
+    CALLBACK_NAME = 'influx'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self):
+        super(CallbackModule, self).__init__()
+
+        self.start_time = datetime.utcnow()
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
+        super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
+
+        self.influxdb_addr = self.get_option('influxdb_addr')
+        self.influxdb_db = self.get_option('influxdb_db')
+        self.influxdb_username = self.get_option('influxdb_username')
+        self.influxdb_password = self.get_option('influxdb_password')
+        self.influxdb_tls = self.get_option('influxdb_tls')
+        self.validate_certs = self.get_option('influxdb_validate_cert')
+
+        if self.influxdb_addr is None:
+            self.disabled = True
+            self._display.warning('InfluxDB address was not provided. The '
+                                  'InfluxDB address can be provided using '
+                                  'the `INFLUXDB_ADDR` environment '
+                                  'variable. Telemetry data not captured.')
+
+        if self.influxdb_db is None:
+            self.disabled = True
+            self._display.warning('InfluxDB database name was not provided. The '
+                                  'InfluxDB database name can be provided using '
+                                  'the `INFLUXDB_DB` environment '
+                                  'variable. Telemetry data not captured.')
+
+        self.influx_constructed_url = 'https://%s/write?db=%s' % (self.influxdb_addr, self.influxdb_db)
+
+        if not self.influxdb_tls:
+            self.influx_constructed_url = 'http://%s/write?db=%s' % (self.influxdb_addr, self.influxdb_db)
+
+    def send_msg(self, role_name, duration):
+        payload = 'role_duration,role=%s duration=%s' % (self.playbook_name, duration)
```

Comment:
Is there no need to escape special characters like spaces in `self.playbook_name`?